src/main/kotlin/org/rust/ide/docs/DocumentationUtils.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.docs
+
+import com.intellij.codeInsight.documentation.DocumentationManagerUtil
+import com.intellij.psi.PsiElement
+import org.rust.ide.utils.escaped
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+
+internal fun PsiElement.generateDocumentation(buffer: StringBuilder, prefix: String = "", suffix: String = "") {
```

Comment:
Hm, indeed.
Maybe drop `DocumentataionUtils` at all and move all functions into `RsDocumentationProvider` as private functions? It'll make `RsDocumentationProvider.kt` bigger but allow not to provide internal methods into global namespace.